### PR TITLE
HTLC API Additions

### LINKED ISSUE #1713: HTLC database APIs
**User Story**
As a user / client developer I want database APIs to query for active HTLC objects.
- [ ] One API to list all HTLC objects.
- [x] APIs to by account, perhaps 2 APIs, one query by `from`, one query by `to`. 
- [x] Pagination should be considered.
- [ ] Perhaps by default order by expiration?
- [x] Perhaps let `get_full_accounts` API return HTLC objects as well.
- [ ] Best have CLI commands as well.

UI work is blocked by this.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by your request. Please tick at least one box.
- [x] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [x] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [ ] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [ ] UX (the User Experience)
- [ ] Other (please add below)

**Additional Context (optional)**
Add any other context about your request here.

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Feature Request
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

### PR DESCRIPTION
**Pull request by jmjatlanta**
This includes the changes of PR #1729, along with fixes requested in that PR.

In short, this PR fixes #1713

### CODE REVIEW

File: libraries/app/application.cpp [hunk 1]
```diff
@@ -340,6 +340,9 @@ void application_impl::set_api_limit() {
 	if(_options->count("api-limit-get-key-references")){
 		_app_options.api_limit_get_key_references = _options->at("api-limit-get-key-references").as<uint64_t>();
 	}
+   if(_options->count("api-limit-get-htlc-by")) {
+      _app_options.api_limit_get_htlc_by = _options->at("api-limit-get-htlc-by").as<uint64_t>();
```
Comment (jmjatlanta) [inline libraries/app/application.cpp]: I would have liked a more descriptive name for this, and have it cover both "to" and "from", but I could not think of one.

File: tests/tests/htlc_tests.cpp [hunk 2]
```diff
@@ -832,4 +832,158 @@ try {
...
+
+   auto htlcs_dan = db_api.get_htlc_by_to(dan.name, graphene::chain::htlc_id_type(0), 100);
+   BOOST_CHECK_EQUAL( htlcs_dan.size(), 1 );
+   BOOST_CHECK_EQUAL( htlcs_dan[0].id.instance(), 2 );
+
+   auto full = db_api.get_full_accounts({alice.name}, false);
+   BOOST_CHECK_EQUAL( full[alice.name].htlcs.size(), 3 );
+
+   full = db_api.get_full_accounts({bob.name}, false);
+   BOOST_CHECK_EQUAL( full[alice.name].htlcs.size(), 0 );
```
Comment (pmconrad) [inline tests/tests/htlc_tests.cpp]: This should be `bob.name`, and the count shouldn't be `0`.

File: tests/tests/htlc_tests.cpp [hunk 3]
```diff
@@ -832,4 +832,158 @@ try {
...
+   graphene::chain::htlc_id_type alice_htlc_id_carl;
+   graphene::chain::htlc_id_type alice_htlc_id_dan;
+
+   generate_block();
+   set_expiration( db, trx );
+   trx.clear();
+   // alice puts a htlc contract to bob
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 2 coins to Bob");
```
Comment (pmconrad) [inline tests/tests/htlc_tests.cpp]: Actually she's transferring 3 not 2 (same in the other cases)

**Comment by oxarbitrage** [comment]
nice work @jmjatlanta

**Pull request merged**